lms/services/moodle.py

```diff
@@ -137,14 +137,17 @@ def list_files(self, course_id: int):
     def file_exists(self, file_id) -> bool:
         """Check if the file exists in the course."""
         # Moodle file IDs are URLs, but they need the token to be accessible
-        response = self._http.request("HEAD", f"{file_id}&token={self.token}")
-        # Moodle API doesn't use status codes, we can't rely on that.
-        # We don't want to download the full file so we'll do a HEAD request and assume:
-        #   - JSON response, it's an error response
-        #   - Anything else, it's  the file we are after
-
-        LOG.info("Headers from Moodle file check %s", response.headers)
-        return not response.headers["content-type"].startswith("application/json")
+        response = self._http.request(
+            "GET", f"{file_id}&token={self.token}", headers={"Range": "bytes=0-1024"}
```

Comment:
It turns out my information about Adobe requiring "%PDF" in the first 1024 bytes of the file is out of date, per https://helpx.adobe.com/acrobat/kb/pdf-error-1015-11001-update.html. I expect other PDF viewers are still more liberal though.